File: contest/typical90X/typical90X.cpp
#include <bits/stdc++.h>
using namespace std;

using ll = long long;

ll N, K, A[1000], B[1000];

int main(){
    cin >> N >> K;

    int count = 0;
    for (int i=0; i<N; i++) {
        cin >> A[i];
    }
    for (int i=0; i<N; i++) {
        cin >> B[i];
    }
    for (int i=0; i<N; i++) {
        count += abs(A[i] - B[i]);
    }

    if (K >= count && (K-count)%2LL == 0) cout << "Yes" << endl;
    else cout << "No" << endl;
    return 0;
}

File: contest/typical90X/typical90X.py
import sys

input = lambda: sys.stdin.readline().rstrip()
N, K = map(int, input().split())
A = list(map(int, input().split()))
B = list(map(int, input().split()))
count = sum([abs(a - b) for a, b in zip(A, B)])
print("Yes" if K >= count and (K - count) % 2 == 0 else "No")
